src/AspNetCore.Diagnostics.CachedHealthChecks/CacheableHealthCheckService.cs

```diff
@@ -340,7 +340,13 @@ private async Task<HealthReportEntry> RunCheckAsync(IServiceScope scope, HealthC
                 exception: exception,
                 data: data);
 
-            memoryCache.Set(cacheKey, entry, expiresAt);
+            var memoryCacheEntryOptions = new MemoryCacheEntryOptions
+            {
+                AbsoluteExpirationRelativeToNow = expiresIn,
+                Size = System.Runtime.InteropServices.Marshal.SizeOf(entry)
```

Comment:
This change has absolute no effect on the cache by itself. It must be combined with the `SizeLimit` property of the `MemoryCache` during registration in the DI container.